### PR TITLE
chore: Update SUPPORT_POLICY.rst

### PR DESCRIPTION
**Pull request by ShubhamChaturvedi7**
*Issue #, if available:*

*Description of changes:*
Updates the SUPPORT_POLICY.rst to mark 1.x and 2.x as End-of-Support

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: SUPPORT_POLICY.rst [hunk 1]
```diff
@@ -22,13 +22,13 @@ This table describes the current support status of each major version of the AWS
...
+      - 
     * - 2.x
-      - Maintenance
       - End of Support
-      - 2022-07-15
+      - 
+      - 
     * - 3.x
       - Generally Available
       -
```
Comment (texastony) [inline SUPPORT_POLICY.rst:34]: Suggestion/Blocking: Add the maintenance date. 
This is the soonest we will move it to maintenance, 
6 months from now.
We can miss this this date, and often do. 
But we are supposed to give customer's at least 6 months warning. 
```suggestion
    * - 3.x
      - Generally Available
      - Maintenance
      - 2023-07-23
```

File: SUPPORT_POLICY.rst [hunk 2]
```diff
@@ -22,16 +22,17 @@ This table describes the current support status of each major version of the AWS
...
       - End of Support
-      - 2022-07-15
+      - 
+      - 
     * - 3.x
       - Generally Available
-      -
+      - Maintenance
+      - 2023-07-23
       -
```
Comment (texastony) [inline SUPPORT_POLICY.rst:36]: 4 Columns, 4 `-`. 
```suggestion
```

**Review by texastony — CHANGES_REQUESTED** [review]

File: SUPPORT_POLICY.rst (hunk 2 above)
Comment (ShubhamChaturvedi7) [reply SUPPORT_POLICY.rst:36]: Weird that it added an extra `-`. I just had accepted the suggestion without making any change on my side: https://github.com/aws/aws-dynamodb-encryption-python/pull/518#discussion_r1073097544

Comment (texastony) [reply SUPPORT_POLICY.rst]: Yes, there are limits to what I can "suggest".
The last `-` was too far from the changed content,
so GitHub would not allow me to suggest a change to it. 

But you can always ask GitHub to render the Markdown from your branch,
and then you will know if your table is formatted correctly.

**Review by texastony — APPROVED** [review]

**Pull request merged**